src/methods/method_overloading/hard/q8/MainTest.java

```diff
@@ -0,0 +1,72 @@
+package methods.defining_methods.hard.q8;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import java.util.stream.Stream;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import global.BaseTest;
+import global.tools.CustomAssertions;
+import global.utils.MethodUtil;
+import global.variables.Clause;
+import global.variables.clauses.NewLine;
+import global.variables.clauses.StringLiteral;
+
+public class MainTest extends BaseTest {
+  // Java
```

Comment:
As this question is a Java question, we have to change the question so that students can not hardcode the output. Due to how the scanner works, we will probably have to have them ask for an integer and a double separately, and then call the printType method separately.

Reply:
Also, did you change the original question so that the printType method returns a string instead of printing the type in that method? Because we can check printed output in a method.